From 121b5c8760bf719ce8f4cfadbb45dfc06553163f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:11:08 -0400
Subject: [powerpc] ftrace_caller, _mcount is exported to modules so needs _GLOBAL_TOC()

Message-id: <1410545655-205645-440-git-send-email-dzickus@redhat.com>
Patchwork-id: 94395
O-Subject: [RHEL7 PATCH 439/626] powerpc: ftrace_caller, _mcount is exported to modules so needs _GLOBAL_TOC()
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 5e66684fe4c71e4d62d6a5d313057185ac0890cc
Author: Anton Blanchard <anton@samba.org>
Date:   Fri Apr 4 09:06:33 2014 +1100

    powerpc: ftrace_caller, _mcount is exported to modules so needs _GLOBAL_TOC()

    When testing the ftrace function tracer, I realised that ftrace_caller
    and mcount are called from modules and they both call into C, therefore
    they need the ABIv2 global entry point to establish r2.

    Signed-off-by: Anton Blanchard <anton@samba.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/entry_64.S b/arch/powerpc/kernel/entry_64.S
index 772fdb6..9075dba 100644
--- a/arch/powerpc/kernel/entry_64.S
+++ b/arch/powerpc/kernel/entry_64.S
@@ -1180,7 +1180,7 @@ _GLOBAL(mcount)
 _GLOBAL(_mcount)
  blr
 
-_GLOBAL(ftrace_caller)
+_GLOBAL_TOC(ftrace_caller)
  /* Taken from output of objdump from lib64/glibc */
  mflr r3
  ld r11, 0(r1)
@@ -1204,10 +1204,7 @@ _GLOBAL(ftrace_graph_stub)
 _GLOBAL(ftrace_stub)
  blr
 #else
-_GLOBAL(mcount)
- blr
-
-_GLOBAL(_mcount)
+_GLOBAL_TOC(_mcount)
  /* Taken from output of objdump from lib64/glibc */
  mflr r3
  ld r11, 0(r1)
-- 
1.7.1